edb/server/mng_port/edgecon.pyx

```diff
@@ -2023,3 +2033,54 @@ cdef class EdgeConnection:
         msg.write_len_prefixed_bytes(b'RESTORE')
         self.write(msg.end_message())
         self.flush()
+
+
+@cython.final
+cdef class Timer:
+    def __init__(self) -> None:
+        self._durations: Dict[str, List[float]] = {}
+        self._last_report_timestamp: Dict[str, float] = {}
+        self._threshold_seconds = 300
+
+    @contextlib.contextmanager
+    def timed(self, operation: str):
+        ts_start = time.monotonic()
+        yield
```

Comment:
Do you want to still log if there was an error in `with timed():`?

Reply:
I assumed that exceptional cases don't yield representative timings. You'd like to wrap that with `try:finally:`?

Reply:
What if a bogus client sends a broken query too frequently -- it still consumes resources for that (compiler, etc). Do we need to account for those resources or not?

Reply:
Let's add try..finally